default/python/AI/TechsListsAI.py

```diff
@@ -452,6 +453,7 @@ def __init__(self):
             self.economy,
             self.economy,
             self.economy,
+            self.economy,
```

Comment:
Does the order within TechGroup3.economy matter when calling self.economy in TechGroup3A?

Reply:
AFAIK it takes the next entry from economy dict

Reply:
To be sure: that means "yes, the order in TechGroup3 does matter", right?
I'm sorry, I don't know what is a dict or how it works, if it is ordered, if "next" takes something at random, etc.

Reply:
Ah sorry. yes, order does matter. The tech names are in a "list" (using square brackets), which is ordered. The researchAI uses (unordered) sets and lists,  techs_to_add is the relevant one which is uses to issue tech enqueue orders.

 A "dict" is a key-value datatype and I think ordered but not in all pythons (using squibbly brackets).